js/components/main.js

```diff
@@ -102,6 +102,19 @@ class Main extends ImmutableComponent {
     // TODO
   }
 
+  onHamburgerMenu () {
+    let settings = {}
+    const resourceNames = ['httpsEverywhere', 'adblock', 'trackingProtection', 'adInsertion']
+    resourceNames.forEach((name) => {
+      let enabled = this.props.appState.getIn([name, 'enabled'])
+      settings[name] = enabled === undefined ? AppConfig[name].enabled : enabled
+    })
+    // whether the current page is bookmarked. needed to re-initialize the
+    // application menu.
+    settings.bookmarked = this.refs.navBar.bookmarked
```

Comment:
yes the approach in https://github.com/brave/browser-laptop/pull/316/files is better thanks.